### PR TITLE
[lldb][debugserver][NFC] Simplify macOS thread name fetching.

### PR DESCRIPTION
**Pull request by ldm0**
Remove unnecessary `proc_pidinfo` calling.

### CODE REVIEW

File: lldb/tools/debugserver/source/MacOSX/MachThread.h [hunk 1]
```diff
@@ -126,6 +127,8 @@ class MachThread {
 protected:
   static bool GetBasicInfo(thread_t threadID,
                            struct thread_basic_info *basic_info);
+  static bool GetExtendedInfo(thread_t threadID,
+                              struct thread_extended_info *extended_info);
 
   bool GetIdentifierInfo();
```
Comment (JDevlieghere) [inline lldb/tools/debugserver/source/MacOSX/MachThread.h:133]: `MachThread::GetIdentifierInfo` is now gone, right?

Comment (ldm0) [reply lldb/tools/debugserver/source/MacOSX/MachThread.h]: Thanks, I didn't notice that.

**Review by JDevlieghere — APPROVED** [review]
LGTM if this makes sense to @jasonmolenda

File: lldb/tools/debugserver/source/MacOSX/MachThread.cpp [hunk 2]
```diff
@@ -265,6 +264,25 @@ bool MachThread::GetBasicInfo(thread_t thread,
...
 }
 
+struct thread_extended_info *MachThread::GetExtendedInfo() {
+  if (MachThread::GetExtendedInfo(m_mach_port_number, &m_extended_info))
+    return &m_extended_info;
+  return NULL;
+}
+
+bool MachThread::GetExtendedInfo(thread_t thread,
+                                 struct thread_extended_info *basicInfoPtr) {
```
Comment (jimingham) [inline lldb/tools/debugserver/source/MacOSX/MachThread.cpp:274]: It's confusing to have a variable called "basicInfoPtr" actually be an "extended_info"...  Can you change this name?

**Comment by jimingham** [comment]
I asked some folks who know and the proc_pidinfo call and the extended_thread_info read the same data, so this should be NFC...

File: lldb/tools/debugserver/source/MacOSX/MachThread.cpp (hunk 2 above)
Comment (ldm0) [reply lldb/tools/debugserver/source/MacOSX/MachThread.cpp]: fixed

**Comment by ldm0** [comment]
> I asked some folks who know and the proc_pidinfo call and the extended_thread_info read the same data, so this should be NFC...

Yes, this PR does not intend to change the behavior: [`thread_info`](https://github.com/apple/darwin-xnu/blob/a1babec6b135d1f35b2590a1990af3c5c5393479/osfmk/kern/thread.c#L1984), [`proc_pidinfo`](https://github.com/apple/darwin-xnu/blob/2ff845c2e033bd0ff64b5b6aa6063a1f8f65aa32/osfmk/kern/bsd_kern.c#L1076). Just a drive-by optimization to reduce syscalls.

**Review by jasonmolenda — APPROVED** [review]
Looks good to me, thanks for putting this PR together.  I can merge the PR for you, if you don't have permissions to do that on llvm-project currently.

**Pull request merged**